### ISSUE TITLE
[Feature Request]: As a Developer, I would like to import types from `@carbon/react`

### ISSUE THREAD

**Issue by digiacomo-a** [carbon-design-system/carbon#16436]
### The problem

Users are often using reusable components to render Tags with different colors (type) based on some status.
To achieve this, a TagType enum needs to be defined because we can not import it from carbon/react.

For instance:

```
enum TagType {
        red = 'red'
        green....
        ... 
}

interface CustomInterface {
  type: TagType,
  title: string,
}
```

### The solution

Could be nice to define and export a `TagType` containing all supported types and then import it so that we can use it in our interfaces, couple of examples below.

### Examples

Example of an interface for Props, with tagType
```
import {TagType} from '@carbon/react';

interface CustomInterface {
  tagType: TagType,
  title: string,
}

`<MyCustomCardWithTag tagType={TagType.red} title="Example" />`

```

### Business priority

Low Priority = release date is not dependent on fix or not upcoming

### Available extra resources

I'm happy to contribute if this change can be useful.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by tay1orjones** [comment]
Yes this isn't exported from the root, but it's exported from the root source file and so you can import them like this:

```js
import { TYPES } from '@carbon/react/es/components/Tag/Tag';
```

I'm not opposed to adding these to the root exports, but we won't be able to do that until everything is converted to .ts, https://github.com/carbon-design-system/carbon/issues/12513, https://github.com/carbon-design-system/carbon/issues/16360

For now I'll add this as a subitem of https://github.com/carbon-design-system/carbon/issues/16154

**Comment by Gururajj77** [comment]
I think we have to have all the export types to be named

https://github.com/microsoft/TypeScript/pull/35200

**Comment by tay1orjones** [comment]
If we upgrade to TS v5 we'd be able to just use

```
export type * from '@carbon/icons-react
```

in here

https://github.com/carbon-design-system/carbon/blob/51d41fa647bc8d31b428f5c485329664879d5a74/packages/react/icons/src/index.ts#L8

https://github.com/microsoft/TypeScript/pull/52217

**Comment by tay1orjones** [comment]
For now we'll export everything by name in the root/index.js while we wait for wildcard exports via TS5, #16827
